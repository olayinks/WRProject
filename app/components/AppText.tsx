import React, { ReactNode } from "react";
import { Text, StyleSheet, Platform, TextStyle } from "react-native";

interface Prop{
    children?:ReactNode,
    style?: TextStyle
}

function AppText({ children, style }:Prop): JSX.Element {
  return <Text style={[styles.text, style]}>{children}</Text>;
}

const styles = StyleSheet.create({
  text: {
    fontSize: 18,
    fontFamily: Platform.OS === "android" ? "Roboto" : "san-francisco",
  },
});

export default AppText;
