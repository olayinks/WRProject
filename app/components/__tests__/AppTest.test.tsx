import React from "react";
import renderer from "react-test-renderer";
import AppText from "../AppText";

test("renders correctly", () => {
  const tree = renderer.create(<AppText />).toJSON();
  expect(tree).toMatchSnapshot();
});
