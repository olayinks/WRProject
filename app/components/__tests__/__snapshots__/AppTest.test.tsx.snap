// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "san-francisco",
        "fontSize": 18,
      },
      undefined,
    ]
  }
/>
`;
